pkgs/by-name/al/alist/update.nix

```
@@ -0,0 +1,50 @@
+{
+  writeShellApplication,
+  lib,
+  nix,
+  nix-update,
+  curl,
+  coreutils,
+  gnused,
+  jq,
+  common-updater-scripts,
+}:
+
+lib.getExe (writeShellApplication {
+  name = "update-alist";
+  runtimeInputs = [
+    coreutils
+    curl
+    gnused
+    jq
+    nix
+    common-updater-scripts
+    nix-update
+  ];
+
+  text = ''
+    set -euo pipefail
+
+    # get old info
+    oldVersion=$(nix-instantiate --eval --strict -A "alist.version" | jq -e -r)
+
+    get_latest_release() {
+        local repo=$1
+        curl --fail ''${GITHUB_TOKEN:+ -H "Authorization: bearer $GITHUB_TOKEN"} \
+             -s "https://api.github.com/repos/AlistGo/$repo/releases/latest" | jq -r ".tag_name"
+    }
+
+    version=$(get_latest_release "alist")
+    version="''${version#v}"
+    webVersion=$(get_latest_release "alist-web")
+
+    if [[ "$oldVersion" == "$version" ]]; then
+        echo "Already up to date!"
+        exit 0
+    fi
+
+    nix-update alist --version "$version"
+
+    update-source-version alist "$webVersion" --source-key=web --version-key=webVersion
```

```suggestion
    update-source-version alist "$webVersion" --source-key=web --version-key=webVersion --ignore-same-hash
```

This fixes the following error:

```console
Going to be running update for following packages:
 - alist-3.39.2

Press Enter key to continue...

Running update for:
 - alist-3.39.2: UPDATING ...
 - alist-3.39.2: ERROR

--- SHOWING ERROR LOG FOR alist-3.39.2 ----------------------

update-source-version: error: Both the old and new source hashes of 'alist.web' were equivalent. Please fix the package's source URL to be dependent on '${version}'!


--- SHOWING ERROR LOG FOR alist-3.39.2 ----------------------
The update script for alist-3.39.2 failed with exit code 1
```

Please adding a comment about why `--ignore-same-hash` is necessary.